src/lib.rs

```diff
@@ -41,1346 +53,988 @@ use std::{
-                                value += &(coeff * poly.evaluate(point));
-                                lc.push((coeff, label.to_string().into()));
-                            }
-                        }
-                    }
-                    values.insert((label.clone(), point), value);
-                    if !lc.is_empty() {
-                        linear_combinations.push(lc);
-                        // Insert query
-                        query_set.insert((label.clone(), (format!("{}", i), point)));
-                    }
-                }
-            }
-            if linear_combinations.is_empty() {
-                continue;
-            }
-            println!("Generated query set");
-            println!("Linear combinations: {:?}", linear_combinations);
-
-            let mut fs_rng = setup_test_fs_rng::<F, PC>();
-            let proof = PC::open_combinations(
-                &ck,
-                &linear_combinations,
-                &polynomials,
-                &comms,
-                &query_set,
-                &mut fs_rng,
-                &rands,
-                Some(rng),
-            )?;
-            assert!(proof.is_valid());
-
-            println!("Generated proof");
-
-            let mut fs_rng = setup_test_fs_rng::<F, PC>();
-            let result = PC::check_combinations(
-                &vk,
-                &linear_combinations,
-                &comms,
-                &query_set,
-                &values,
-                &proof,
-                &mut fs_rng,
-            )?;
+        let mut combined_commitment = Self::Commitment::zero();
+        let mut combined_value = G::ScalarField::zero();
 
-            if !result {
-                println!(
-                    "Failed with {} polynomials, num_points_in_query_set: {:?}",
-                    num_polynomials, num_points_in_query_set
-                );
-                println!("Degree of polynomials:",);
-                for poly in polynomials {
-                    println!("Degree: {:?}", poly.degree());
-                }
+        let lambda: G::ScalarField = fs_rng.squeeze_128_bits_challenge();
+        let mut cur_challenge = G::ScalarField::one();
+
+        let labeled_commitments_iter = labeled_commitments.into_iter();
+        let values_iter = values.into_iter();
+
+        for (labeled_commitment, value) in labeled_commitments_iter.zip(values_iter) {
+            let commitment = labeled_commitment.commitment();
+            combined_value += &(cur_challenge * &value);
+
+            if cur_challenge == G::ScalarField::one() {
+                combined_commitment = commitment.clone();
+            } else {
+                combined_commitment += &(commitment.clone() * &cur_challenge);
             }
-            assert!(
-                result,
-                "proof was incorrect, equations: {:#?}",
-                linear_combinations
-            );
+
+            cur_challenge *= &lambda;
```

Comment:
Just for consistency: we could use LC combine here, too.

Reply:
Done